graphistry/umap_utils.py

```diff
@@ -432,6 +430,9 @@ def umap(
         else:
```

Comment:
I think this needs to move up before the lazy initialization too, right, because otherwise all the self.suffix and lazy init calls will mutate `self` (and those should be called on `res` instead of `self`)

@silkspace any reason that's not the case?

Reply:
Working with Daniel now, the engine should be resolved inside the constructor, and bound to the res.engine after the fact.

Reply:
@silkspace but isn't that unsafe?

```python
g1 = g.umap(engine='cuml')
g2 = g.umap(engine='umap')
assert g1.engine != g2.engine
```

Reply:
I see where I was confused, yes, i was commenting on the previous 

`res.engine = engine`

which was changed to 

`res.engine = resolve_engine(engine)`

But you were referring to the `lazy_umap_init` code

Reply:
put
```
res.umap_lazy_init(engine=engine)  after the code blocks
        if inplace:
            res = self
        else:
            res = self.bind()
```
that will ensure that all settings are placed in res, no matter what.
also then there is no need to have
`res.engine = resolve_umap_engine(engine)` after the fact, as it has been set in self which is res

(confused on what part we were referring), all setting should be set in constructor, which in this setting happens in `umap_lazy_init(..)`